### PR TITLE
Quaternion: Added world-to-local methods

### PR DESCRIPTION
**Pull request by WestLangley**
As proposed in https://github.com/mrdoob/three.js/issues/13704#issuecomment-684058704.

I'll add docs if there is interest.

/ping @donmccurdy

### CODE REVIEW

File: src/math/Quaternion.js [hunk 1]
```diff
@@ -384,6 +384,18 @@ class Quaternion {
 
 	}
 
+	localToWorld( object ) {
```
Comment (Mugen87) [inline src/math/Quaternion.js:387]: I personally think that math primitives like `Quaternion` or `Vector3` should not know about stuff like local and world coordinates systems. These terms and thus these methods belong to `Object3D`, IMO.

Comment (WestLangley) [reply src/math/Quaternion.js:387]: >math primitives like Quaternion or Vector3 should not know about stuff like local and world coordinates systems.

Fortunately, they don't.

Like wise for
```js
vector.fromBufferAttribute( attribute )
box3.setFromObject( object );
bufferGeometry.setFromObject( object );
convexHull.setFromObject( object );
```

Comment (Mugen87) [reply src/math/Quaternion.js:387]: Well, I've never liked this API style. From an engineering point-of-view, these methods belong elsewhere.

Just want to make clear that it's worthy of discussion if the project keeps following this approach.

Comment (WestLangley) [reply src/math/Quaternion.js:387]: I think you are beginning with a false premise. The object class is not imported into these classes.

Comment (WestLangley) [reply src/math/Quaternion.js:387]: Also, as I have said multiple times, it is `vector.applyMatrix( m )`, not `matrix.applyToVector( v )`.

Similarly, `quaternion.worldToLocal( object )`; not `object.worldToLocal( quaternion )`.

In any event, we need to first see if this PR is desired. :-)

Comment (Mugen87) [reply src/math/Quaternion.js:387]: It's not only about the imports. It's the fact the math methods actually work with classes like `Object3D` in their implementations. 

Anyway, it's not my intention to block your contribution. This PR just seemed a good opportunity to highlight something that bothers me for a while now 😇 .

Comment (gkjohnson) [reply src/math/Quaternion.js:387]: I think this is a nice addition -- I think I'd expect it to be on Object3D, though, next to the vector equivalents:

```js
object.localToWorldRotation( quat );
```

Or maybe the existing `localToWorld` and `worldToLocal` functions could be modified to support taking either a quaternion or vector parameter:

```js
object.localToWorld( quat );
object.localToWorld( vec );
```

Comment (WestLangley) [reply src/math/Quaternion.js:387]: The "vector equivalents" need to be moved to the vector class, where they should have been added in the first place.
```js
vector.localToWorld( obj1 ).worldToLocal( obj2 );
```
not
```js
obj2.worldToLocal( obj1.localToWorld( vector ) );
```

Comment (donmccurdy) [reply src/math/Quaternion.js:387]: @gkjohnson see https://github.com/mrdoob/three.js/issues/13704#issuecomment-684058704 — there is some feeling here that the vector equivalents are in the wrong place. :)

**Comment by WestLangley** [comment]
@donmccurdy Is this the feature you wanted?

**Comment by donmccurdy** [comment]
@WestLangley this provides the API I was hoping for, yes! But I have to admit I'd personally probably not use them, and instead copy/paste a version that doesn't update matrices, for my own projects. Having `math/` functions performing scene graph updates still feels unnecessary, I tend to expect these functions to behave as efficiently as possible. Getting 1x world position and 1x world quaternion shouldn't require computing the world matrix twice, especially in the render loop.

Another idea would be...

```js
quaternion.localToWorld( matrixWorld );
```

... in that case, it may at least be more obvious to users that the matrix should be updated beforehand?

**Comment by WestLangley** [comment]
How about
```js
quaternion.localToWorld( object, updateWorldMatrix );
```
with `updateWorldMatrix` defaulting to `true`.

We would have to add this arg everywhere.

**Comment by WestLangley** [comment]
It seems this is not getting sufficient traction, so closing...

**Pull request closed without merging**